server/models/device_instance.py

```
@@ -664,6 +678,32 @@ def updateDeviceColumn(self, mac, column_name, column_value):
         return result
 
+    def lockDeviceField(self, mac, field_name):
+        """Lock a device field so it won't be overwritten by plugins."""
+        if field_name not in FIELD_SOURCE_MAP:
+            return {"success": False, "error": f"Field {field_name} does not support locking"}
+
+        try:
+            conn = get_temp_db_connection()
+            lock_field(mac, field_name, conn)
+            conn.close()
+            return {"success": True, "message": f"Field {field_name} locked"}
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)